src/main/java/gyro/google/artifactregistry/RepositoryResource.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    /**
+     * The KMS key that’s used to encrypt the contents of the Repository.
+     */
+    public CryptoKeyResource getKey() {
+        return key;
+    }
+
+    public void setKey(CryptoKeyResource key) {
+        this.key = key;
+    }
+
+    /**
+     * The set of labels for this repository.
+     */
+    @Updatable
+    public Map<String, String> getLabels() {
+        if (labels == null) {
+            labels = new HashMap<>();
+        }
+
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * The ID of the repository.
+     */
+    @Output
+    @Id
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    @Override
+    public void copyFrom(Repository model) throws Exception {
+        setId(model.getName());
+        setDescription(model.getDescription());
+        setFormat(model.getFormat());
+        setKey(findById(CryptoKeyResource.class, model.getKmsKeyName()));
+        setName(Utils.getRepositoryNameFromId(model.getName()));
+        setLocation(Utils.getLocationFromId(model.getName()));
+        setLabels(model.getLabelsMap());
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        ArtifactRegistryClient client = createClient(ArtifactRegistryClient.class);
+
+        Repository repository = getRepository(client);
+
+        if (repository == null) {
+            return false;
+        }
+
+        copyFrom(repository);
+
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        ArtifactRegistryClient client = createClient(ArtifactRegistryClient.class);
+
+        Repository.Builder builder = Repository.newBuilder().setName(getName())
+            .setFormat(Repository.Format.DOCKER).putAllLabels(getLabels());
+
+        if (!StringUtils.isEmpty(getDescription())) {
+            builder.setDescription(getDescription());
+        }
+
+        if (getKey() != null) {
+            builder.setKmsKeyName(getKey().getId());
+        }
+
+        try {
+            Repository repository = client.createRepositoryAsync(CreateRepositoryRequest.newBuilder()
+                .setParent(getParent())
+                .setRepositoryId(getName()).setRepository(builder.build()).build()).get();
+            setId(repository.getName());
+
+        } catch (ExecutionException ex) {
+            client.awaitTermination(10, TimeUnit.SECONDS);
+            Repository repository = client.listRepositories(getParent()).getPage()
+                .getResponse().getRepositoriesList().stream()
+                .filter(r -> Utils.getRepositoryNameFromId(r.getName()).equals(getName())).findFirst().orElse(null);
+            if (repository == null) {
+                throw ex;
+            } else {
+                setId(repository.getName());
+            }
+        }
```

Comment:
Should we do this blindly or do it on a particular type content of the exception ?